### ISSUE TITLE
Problem with TensorFlow model example with custom Sin operator

### ISSUE THREAD

**Issue by Lucy20211** [tensorflow/tensorflow#47462]
Hi,

I would like to practice with the TensorFlow model example with custom Sin operator (https://www.tensorflow.org/lite/guide/ops_custom#create_a_tensorflow_model). In particular, I'm interested to repeat the error 
"Error: Some of the operators in the model are not supported by the standard TensorFlow Lite runtime...... Here is
a list of operators for which you will need custom implementations: Sin." 

How can I do this using, for example, tf.lite.TFLiteConverter.from_concrete_functions()?
Thanks in advance.

**Comment by abattery** [comment]
@Lucy20211 you need to add the `converter.allow_custom_ops = True` line to enable custom ops.

**Comment by Lucy20211** [comment]
Thanks @abattery for you reply, but before using the line you suggested I would just like to repeat the above error. Since the model in the example does not consist of layers, how can I tell the converter to try to convert in a TFLite model the model that uses the Sin operator?

**Comment by abattery** [comment]
Could you create a gist for the problem? If the above flag is on, the above error message will become an warning and the conversion will be succeded. If not, there is something wrong happening but it is rare since the custom operator enabling is quite stable for a long time.

**Comment by Lucy20211** [comment]
I haven't used the converter yet, I only copy and paste this part of the example code:

import tensorflow as tf
 
# Define training dataset and variables
x = [-8, 0.5, 2, 2.2, 201]
y = [-0.6569866 ,  0.99749499,  0.14112001, -0.05837414,  0.80641841]
offset = tf.Variable(0.0)

# Define a simple model which just contains a custom operator named `Sin`
@tf.function
def sin(x):
  return tf.sin(x + offset, name="Sin")

  # Train model
optimizer = tf.optimizers.Adam(0.01)
def train(x, y):
    with tf.GradientTape() as t:
      predicted_y = sin(x)
      loss = tf.reduce_sum(tf.square(predicted_y - y))
    grads = t.gradient(loss, [offset])
    optimizer.apply_gradients(zip(grads, [offset]))

for i in range(1000):
    train(x, y)

print("The actual offset is: 1.0")
print("The predicted offset is:", offset.numpy())


Then I would like the above error to appear, but I don't know how I should use the converter. I'm new to TensorFlow, so I'd like to repeat the example step by step.

**Comment by abattery** [comment]
Yeah, I confirmed the custom op example is outdated. Sorry for that. tf.Sin op is now a part of the TFLite builtin op set. You can use tf.RandomUniform operator to produce the custom op.

For example,

```
  @tf.function(input_signature=[tf.TensorSpec(shape=[None], dtype=tf.float32)])
  def sin(x):
    return tf.random.uniform([], 0, 10, dtype=tf.float32)

  converter = tf.lite.TFLiteConverter.from_concrete_functions([sin.get_concrete_function()])
  converter.allow_custom_ops = True
  tflite_model = converter.convert()
```

**Comment by Lucy20211** [comment]
Ok, therefore can I convert an entire trained model in a TFLite model with:

`tf.lite.TFLiteConverter.from_concrete_functions([my_custom_operator.get_concrete_function()])?` 

I mean, shouldn't I try to convert before the trained model somehow, which contains the custom operator that cannot be detected, then use these lines (since the error should appear):

`converter = tf.lite.TFLiteConverter.from_concrete_functions([my_custom_operator.get_concrete_function()])
converter.allow_custom_ops = True ` ?

 By doing so, I thought I could obtain this error at the end:

`Error:
Didn't find custom operator for name 'my custom operator'
Registration failed.`


Sorry, I'm a little confused on this point.

**Comment by Lucy20211** [comment]
After `Error: Didn't find custom operator for name 'my custom operator' Registration failed.` I'd create and register the custom operator.

I would like to practice this simple example in order to later try to implement a 1D convolution operator for a Temporal Convolutional Network, as this operator should not yet be supported by TensorFlow Lite.

**Comment by abattery** [comment]
If the converter does not throw any errors related to the custom operator, it succeded on the conversion with the given TFLite builtin operator, which is a good sign.

**Comment by Lucy20211** [comment]
Okay, I'll try to follow your suggestions, thank you very much @abattery.
